src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/ReadPropertiesStep.java

```diff
@@ -0,0 +1,148 @@
+ * Reads java properties formatted files and texts into a map.
+ *
+ * @author Robert Sandell &lt;rsandell@cloudbees.com&gt;.
+ */
+public class ReadPropertiesStep extends AbstractStepImpl {
+    private String file;
+    private String text;
+    private Map<String, Object> defaults;
+
+    @DataBoundConstructor
+    public ReadPropertiesStep() {
+    }
+
+    /**
+     * The path to a file in the workspace to read the properties from.
+     *
+     * @return te path
+     */
+    public String getFile() {
+        return file;
+    }
+
+    /**
+     * The path to a file in the workspace to read the properties from.
+     *
+     * @param file the path
+     */
+    @DataBoundSetter
+    public void setFile(String file) {
+        this.file = file;
+    }
+
+    /**
+     * A String containing properties formatted data.
+     *
+     * @return text to parse
+     */
+    public String getText() {
+        return text;
+    }
+
+    /**
+     * A String containing properties formatted data.
+     *
+     * @param text text to parse
+     */
+    @DataBoundSetter
+    public void setText(String text) {
+        this.text = text;
+    }
+
+    /**
+     * Default key/values to populate the map with before parsing.
+     *
+     * @return the defaults
+     */
+    public Map<String, Object> getDefaults() {
+        return defaults;
+    }
+
+    /**
+     * Default key/values to populate the map with before parsing.
+     *
+     * @param defaults the defaults
+     */
+    @DataBoundSetter
+    public void setDefaults(Map<String, Object> defaults) {
+        this.defaults = defaults;
+    }
+
+    @Extension
+    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+
+        public DescriptorImpl() {
+            super(ReadPropertiesStepExecution.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "readProperties";
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Read properties from files in the workspace or text.";
+        }
+
+        @Override
+        public Step newInstance(Map<String, Object> arguments) throws Exception {
+            ReadPropertiesStep step = new ReadPropertiesStep();
+            if(arguments.containsKey("file")) {
+                Object file = arguments.get("file");
+                if (file != null) {
+                    step.setFile(file.toString());
+                }
+            }
+            if (arguments.containsKey("text")) {
+                Object text = arguments.get("text");
+                if (text != null) {
+                    step.setText(text.toString());
+                }
+            }
+            if (arguments.containsKey("defaults")) {
+                Object defaults = arguments.get("defaults");
+                if (defaults != null && defaults instanceof Map) {
+                    step.setDefaults((Map)defaults);
+                }
+            }
+            return step;
```

Comment:
As it's implemented now you could have something like:

´´´
node {
    ...
    def props =  readProperties
    ...
}
´´´

Without parameters, it returns an empty map. Does it makes sense?
I'd propose to make either `file` or `text` required, although it's not possible to implement that logic based on `@DataboundConstructor` and `@DataboundSetter` you could do it in the `newInstance` override.